components/Repositories/RossChart.tsx

```diff
@@ -79,57 +128,31 @@ export default function RossChart({ stats, rangedTotal, isLoading, error, range,
             />
             <Tooltip content={CustomTooltip} filterNull={false} />
             <CartesianGrid vertical={false} strokeDasharray="4" stroke="#E2E8F0" />
-            <Bar dataKey="internal" stackId="a" fill="#1E3A8A" />
-            <Bar dataKey="recurring" stackId="a" fill="#2563EB" />
-            <Bar dataKey="new" stackId="a" fill="#60A5FA" />
+            {filterInternal && <Bar dataKey="internal" stackId="a" fill="#1E3A8A" />}
+            {filterRecurring && <Bar dataKey="recurring" stackId="a" fill="#2563EB" />}
+            {filterOutside && <Bar dataKey="new" stackId="a" fill="#60A5FA" />}
           </BarChart>
         )}
       </ResponsiveContainer>
+
+      <fieldset className="flex flex-row gap-4 w-fit text-sm mx-auto">
+        <h3>Filter by:</h3>
```

Comment:
Refactor (blocking): Use a legend with a fieldset.

```suggestion
        <legend className="contents px-1">Filter by:</legend>
```

If the layout isn't working, you can use

`<fieldset className="grid grid-cols-[auto_1fr] gap-2 items-center text-sm">` instead of `<fieldset className="flex flex-row gap-4 w-fit text-sm mx-auto">`